File: ChatApp/app/src/main/java/top/chilfish/chatapp/helper/LoadFile.java
package top.chilfish.chatapp.helper;

import static top.chilfish.chatapp.Main.AppCONTEXT;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

public class LoadFile {

  // load text file from assets
  public static String assetsString(String fileName) {
    StringBuilder fileString = new StringBuilder();

    try (InputStream inputStream = AppCONTEXT.getAssets().open(fileName)) {
      BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
      String line;
      while ((line = reader.readLine()) != null) {
        fileString.append(line);
      }
      return fileString.toString();
    } catch (Exception e) {
      e.printStackTrace();
    }
    return "";
  }
}

File: ChatApp/app/src/main/java/top/chilfish/chatapp/helper/LoadFile.kt
package top.chilfish.chatapp.helper

import top.chilfish.chatapp.Main
import java.io.BufferedReader
import java.io.InputStreamReader

object LoadFile {
    // load text file from assets
    fun assetsString(fileName: String): String {
        val fileString = StringBuilder()
        try {
            Main.AppCONTEXT!!.assets.open(fileName).use { inputStream ->
                val reader = BufferedReader(InputStreamReader(inputStream))
                var line: String?
                while (reader.readLine().also { line = it } != null) {
                    fileString.append(line)
                }
                return fileString.toString()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return ""
    }
}